http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -137,9 +140,19 @@ class ClusterMonitorActor(val cluster: Cluster,
     * @return ScheduleMonitorPass
     */
   private def handleNotReadyCluster(status: ClusterStatus, instances: Set[Instance]): Future[ClusterMonitorMessage] = {
-    logger.info(s"Cluster ${cluster.projectNameString} is not ready yet (cluster status = $status, instance statuses = ${instances.groupBy(_.status).mapValues(_.size)}). Checking again in ${monitorConfig.pollPeriod.toString}.")
-    persistInstances(instances).map { _ =>
-      ScheduleMonitorPass
+    val currTimeElapsed: FiniteDuration = startupTimings.getOrElse(cluster.id, 0 seconds)
+    logger.info(s"Cluster ${cluster.projectNameString} has taken ${currTimeElapsed.toString} so far, continuing to monitor.")
+
+    if (currTimeElapsed > monitorConfig.creationTimeLimit && status == ClusterStatus.Creating) {
+      logger.info(s"Detected that ${cluster.projectNameString} has been creating too long, time limit is ${monitorConfig.creationTimeLimit}. Failing it.")
+      Metrics.newRelic.incrementCounterIO("createClusterTimeout").unsafeRunSync()
+      handleFailedCluster(ClusterErrorDetails(Code.DEADLINE_EXCEEDED.value,Some(s"Failed to create cluster ${cluster.projectNameString} within ${monitorConfig.creationTimeLimit.toMinutes.toString()} minutes")),instances)
+    } else {
+      startupTimings += (cluster.id -> startupTimings.getOrElse(cluster.id, 0 seconds).+(monitorConfig.pollPeriod))
```

Comment:
We have the monitor start time already - couldn't we get the elapsed time by:
```
val elapsedTime = System.currentTimeMillis - startTime
```
?

I'm not sure we need the `startupTimings` Map.